core/src/ledger_cleanup_service.rs

```diff
@@ -169,6 +189,38 @@ impl LedgerCleanupService {
         Ok(new_root_receiver.try_iter().last().unwrap_or(root))
     }
 
+    /// May perform cleanup expired data in ledger store based on the specified
+    /// parameters.  If `root` - `last_purge_slot` > `purge_interval`, ledger
+    /// cleanup will further check whether there're any data to cleanup based
+    /// on the input `max_ledger_shreds`.  If it does, then cleanup will be
+    /// perfomed, and both `last_purge_slot` and `last_compact_slot` will be
+    /// updated accordingly.
+    ///
+    /// If `root` - `last_purge_slot` <= `purge_interval`, the function will
+    /// simply become no-op and return Ok status.
+    ///
+    /// [`new_root_receiver`]: signal receiver which contains the information
+    ///   about what `Slot` is the current root.
+    /// [`blockstore`]: the storage that stores shreds and their metadata.  The
+    ///   data will later be used for replay.
+    /// [`max_ledger_shreds`]: the number of shreds to keep since the new root.
+    ///   The new root information is carried from `new_root_receiver`.
+    /// [`last_purge_slot`]: an both an input and output parameter indicating
+    ///   the id of the last purged slot.  As an input parameter, it works
+    ///   together with `purge_interval` on whether it is too early to perform
+    ///   ledger cleanup.  As an output parameter, it will be updated if this
+    ///   function actually performs the ledger cleanup.
+    /// [`purge_interval`]: the minimum slot interval between two ledger
+    ///   cleanup.  When the root derived from `new_root_receiver` minus
+    ///   `last_purge_slot` is fewer than `purge_interval`, the function will
+    ///   simply return `Ok` without actually running the ledger cleanup.
+    ///   In this case, `purge_interval` will remain unchanged.
+    /// [`last_compact_slot`]: an output value which indicates the most recent
+    ///   slot which has been cleaned up after this call.  If this parameter is
+    ///   updated after this function call, it means the ledger cleanup has
+    ///   been performed.
+    ///
+    /// Also see `find_slots_to_clean`, `blockstore::purge_slot`.
```

Comment:
I do think referring reader to `blockstore::purge_slots` is appropriate here; you could leave as-is or if you wanted to elaborate a little, maybe "This function finds slots to clean, and passes along to blockstore::purge_slots to do the heavy lifting"

That note aside, I have similar thoughts here as I do the other reviews - I think having every input documented is overkill as some of these are self-explanatory. And, if someone is scanning the docs, they should be able to click through on any non-primitive types (ie Blockstore). 

That being said, let's chat about these briefly. I've looked at this code a ton previously and I might be too familiar to properly value what some of these would provide to someone looking at this for the first time